autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +121,79 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
```

Comment:
Any valid string path? So 'file.custom_extension is valid? I don't think it is.

Reply:
It is just a string indicating a path to the file, which is similar to the case in tabular.

Reply:
I understand that, but in your code you raise an exception if the file name does not end in .tsv or .csv. This is not communicated in this documentation.

Reply:
If the user's data format is not in tsv or csv, or not benchmark dataset supported in gluonnlp, we would raise NotImplementedError.